### PR TITLE
Add update method

### PR DESCRIPTION
**Pull request by Apercu**
My bad, totally forgot that one

### CODE REVIEW

**Comment by Apercu** [comment]
@mariocasciaro bump 😄

File: README.md [hunk 1]
```diff
@@ -97,6 +96,26 @@ const newObj = immutable.set(obj, 'a.c.1', 'fooo')
...
+Updates an object property.
+
+```javascript
+const obj = {
+  a: {
+    b: 1,
+  },
+}
+
+const newObj = immutable.set(obj, ['a', 'b'], v => v + 1)
```
Comment (mariocasciaro) [inline README.md]: immutable.update?

File: test.js [hunk 2]
```diff
@@ -72,6 +72,29 @@ describe('set', function () {
   it('should set at a numeric path', function () {
     expect(op.set([], 0, 'yo')).to.deep.equal(['yo'])
   })
+
```
Comment (mariocasciaro) [inline test.js]: Could you also add a test for the new method in the following sections, just to be sure ti works with the bind mode as well?

* should execute all methods on the bound object
* should throw if an operation is attempted after `value` called

**Comment by Apercu** [comment]
Should be good 😄

**Comment by Apercu** [comment]
Not sure why the coverage is reduced now but at least it passes in node <4

**Comment by thenewvu** [comment]
Personally I prefer `apply` than `update`.

**Comment by Apercu** [comment]
Coveralls going mad 😄 

@mariocasciaro Should be good now right?

**Comment by mariocasciaro** [comment]
Awesome! Looks good to me, @Apercu. Merged.

**Comment by Apercu** [comment]
Cool 🐱 Do you mind cutting a patch and publish it?

**Comment by mariocasciaro** [comment]
Sure. Published to 0.5.2.

**Pull request merged**